From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Tue, 17 Feb 2026 05:35:54 +0000
Subject: [PATCH] Fenix - IronFox settings

Adds support for IronFox-specific UI settings to control various functionality.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
index 75f51971cb20..b445d15c4171 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
@@ -144,6 +144,7 @@ import org.mozilla.fenix.theme.ThemeProvider
 import org.mozilla.fenix.utils.Settings
 import org.mozilla.fenix.utils.isLargeScreenSize
 import org.mozilla.fenix.wallpapers.Wallpaper
+import org.ironfoxoss.ironfox.utils.IFPrefs
 
 private const val RAM_THRESHOLD_MEGABYTES = 1024
 private const val BYTES_TO_MEGABYTES_CONVERSION = 1024.0 * 1024.0
@@ -238,6 +239,7 @@ open class FenixApplication : Application(), Provider, ThemeProvider {
             // Start loading the SharedPreferences file from disk on a background thread immediately.
             applicationScope.launch(IO) {
                 applicationContext.getSharedPreferences(Settings.FENIX_PREFERENCES, MODE_PRIVATE)
+                IFPrefs.setPrefs(applicationContext)
             }
 
             // Allow overriding secret settings with values specified in `local.properties` at build time.
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
index aff32cbcd7f3..23e277590c11 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Core.kt
@@ -150,6 +150,7 @@ import org.mozilla.fenix.translations.TranslationsEnabledSettings
 import org.mozilla.fenix.utils.Settings.DeleteDownloadBehavior
 import org.mozilla.fenix.utils.getUndoDelay
 import org.mozilla.geckoview.GeckoRuntime
+import org.ironfoxoss.ironfox.utils.GeckoSettingsBridge
 
 /** Component group for all core browser functionality. */
 @Suppress("LargeClass")
@@ -262,6 +263,7 @@ class Core(
                 runtime = geckoRuntime,
             )
             .also {
+                GeckoSettingsBridge.setIronFoxPrefs(context, it)
                 WebCompatFeature.install(it)
             }
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
index 70b91774b0ec..2d21d2447b87 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
@@ -90,6 +90,7 @@ import org.mozilla.fenix.settings.account.AccountUiView
 import org.mozilla.fenix.snackbar.FenixSnackbarDelegate
 import org.mozilla.fenix.snackbar.SnackbarBinding
 import org.mozilla.fenix.utils.Settings
+import org.ironfoxoss.ironfox.utils.FenixSettingsVisibilityManager
 
 /** Main settings screen. */
 @Suppress("LargeClass", "TooManyFunctions")
@@ -670,6 +671,20 @@ class SettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFragment
         setupTrackingProtectionPreference(settings)
         setupDnsOverHttpsPreference(settings)
         setupEmailMaskPreference(settings, requireComponents)
+        setupIronFoxPreferences()
+    }
+
+    internal fun setupIronFoxPreferences() {
+        requirePreference<Preference>(R.string.pref_key_if_preferences).apply {
+            setOnPreferenceClickListener {
+                val directions =
+                    SettingsFragmentDirections
+                        .actionSettingsFragmentToIronFoxSettingsFragment()
+                requireView().findNavController().navigate(directions)
+                true
+            }
+        }
+        FenixSettingsVisibilityManager.SettingsFragment(requireContext(), this)
     }
 
     private val setToDefaultPromptRequestLauncher: ActivityResultLauncher<Intent> =
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 83e0f40483b7..d1f203994c07 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -4,6 +4,7 @@
 
 package org.mozilla.fenix.utils
 
+import org.ironfoxoss.ironfox.utils.IronFoxSettings
 import android.accessibilityservice.AccessibilityServiceInfo.CAPABILITY_CAN_PERFORM_GESTURES
 import android.appwidget.AppWidgetManager
 import android.content.Context
@@ -171,6 +172,8 @@ class Settings(
         private const val CLOUDFLARE_URI = "https://mozilla.cloudflare-dns.com/dns-query"
     }
 
+    val ironfox = IronFoxSettings(appContext)
+
     private val logger = Logger("Settings")
 
     @VisibleForTesting
diff --git a/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml b/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
index 440da531952f..d4c4e78d2887 100644
--- a/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
+++ b/mobile/android/fenix/app/src/main/res/navigation/nav_graph.xml
@@ -692,6 +692,13 @@
             app:exitAnim="@anim/slide_out_left"
             app:popEnterAnim="@anim/slide_in_left"
             app:popExitAnim="@anim/slide_out_right" />
+        <action
+            android:id="@+id/action_settingsFragment_to_ironFoxSettingsFragment"
+            app:destination="@id/ironFoxSettingsFragment"
+            app:enterAnim="@anim/slide_in_right"
+            app:exitAnim="@anim/slide_out_left"
+            app:popEnterAnim="@anim/slide_in_left"
+            app:popExitAnim="@anim/slide_out_right" />
         <action
             android:id="@+id/action_settingsFragment_to_trackingProtectionFragment"
             app:destination="@id/trackingProtectionFragment"
@@ -1034,6 +1041,10 @@
             app:argType="string"
             app:nullable="true" />
     </fragment>
+    <fragment
+        android:id="@+id/ironFoxSettingsFragment"
+        android:name="org.mozilla.fenix.settings.IronFoxSettingsFragment"
+        android:label="@string/if_preferences" />
     <fragment
         android:id="@+id/trackingProtectionFragment"
         android:name="org.mozilla.fenix.settings.TrackingProtectionFragment">
diff --git a/mobile/android/fenix/app/src/main/res/xml/preferences.xml b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
index 1b720c0285de..62c542f548a0 100644
--- a/mobile/android/fenix/app/src/main/res/xml/preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
@@ -6,12 +6,12 @@
 <androidx.preference.PreferenceScreen xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto">
 
-    <androidx.preference.Preference
+    <!-- <androidx.preference.Preference
         android:key="@string/pref_key_sign_in"
         android:layout="@layout/sign_in_preference"
         android:summary="@string/preferences_sign_in_description"
         android:title="@string/preferences_sign_in"
-        app:allowDividerBelow="false" />
+        app:allowDividerBelow="false" /> -->
 
     <androidx.preference.PreferenceCategory
         android:key="@string/pref_key_account_category"
@@ -20,6 +20,10 @@
         android:layout="@layout/preference_category_no_icon_style">
 
         <org.mozilla.fenix.settings.account.AccountPreference
+            android:icon="@drawable/mozac_ic_avatar_circle_24"
+            android:key="@string/pref_key_sign_in" />
+
+         <org.mozilla.fenix.settings.account.AccountPreference
             android:icon="@drawable/mozac_ic_avatar_circle_24"
             android:key="@string/pref_key_account" />
 
@@ -233,6 +237,7 @@
         <androidx.preference.SwitchPreferenceCompat
             android:defaultValue="false"
             android:key="@string/pref_key_enable_gecko_logs"
+            android:summary="@string/restart_required_summary"
             app:iconSpaceReserved="false"
             android:title="@string/preferences_enable_gecko_logs" />
 
@@ -242,6 +247,17 @@
             android:title="@string/preferences_local_network_access" />
     </PreferenceCategory>
 
+    <androidx.preference.PreferenceCategory
+        android:layout="@layout/preference_category_no_icon_style"
+        android:selectable="false"
+        android:title="@string/if_title"
+        app:iconSpaceReserved="false">
+        <androidx.preference.Preference
+            android:key="@string/pref_key_if_preferences"
+            android:title="@string/if_preferences"
+            app:iconSpaceReserved="false" />
+    </androidx.preference.PreferenceCategory>
+
     <androidx.preference.PreferenceCategory
         android:title="@string/preferences_category_about"
         app:iconSpaceReserved="false"
-- 
